ksql-parser/src/main/java/io/confluent/ksql/parser/tree/TableElement.java

```diff
@@ -63,11 +65,29 @@ public TableElement(
       final Namespace namespace,
       final ColumnName name,
       final Type type
+  ) {
+    this(location, namespace, name, type, Optional.empty());
+  }
+
+  /**
+   * @param location the location in the SQL text.
+   * @param namespace  indicates if the element is part of the key or value.
+   * @param name the name of the element.
+   * @param type the sql type of the element.
+   * @param source the data source the element belongs to
+   */
+  public TableElement(
+      final Optional<NodeLocation> location,
+      final Namespace namespace,
+      final ColumnName name,
+      final Type type,
+      final Optional<SourceName> source
```

Comment:
`TableElement` is the pojo that elements in the CT/CS schema are deserialized to, right?  e.g. `CREATE TABLE FOO (X INT, Y DOUBLE) ...` would result in a two `TableElement`s for `X` and `Y`.

It feels wrong to extend the model pojo to hold things that our syntax doesn't need.

Can you explain _why_ you need an optional source name, because at the moment I don't understand the reasoning.

Reply:
From an offline chat, this is needed because the `LogicalSchemaDeserializer` uses the `SchemaParser` which returns `TableElements`.  

I think a better approach would be to keep `TableElement` as it is, (with no source), as that is what the contract is for CS/CT statements. Instead, add a new type to capture Schema elements.